File: Carros Objective-C/Carros Objective-C/Model/Utils/Prefs.m
//
//  Prefs.m
//  Carros Objective-C
//
//  Created by Antonio Lopes on 8/10/16.
//  Copyright © 2016 Antonio Lopes. All rights reserved.
//

#import "Prefs.h"

@implementation Prefs

+ (void) setObjectForKey:(id)valor withChave:(NSString *)chave {
    NSUserDefaults * prefs = [NSUserDefaults standardUserDefaults];
    [prefs setObject:valor forKey:chave];
    [prefs synchronize];
}

+ (void) setIntegerForKey:(NSInteger)valor withChave:(NSString *)chave {
    NSUserDefaults * prefs = [NSUserDefaults standardUserDefaults];
    [prefs setInteger:valor forKey:chave];
    [prefs synchronize];
}

+ (NSObject *) getObjectForKey:(NSString *)chave {
    NSUserDefaults * prefs = [NSUserDefaults standardUserDefaults];
    return [prefs objectForKey:chave];
}

@end

File: Carros Swift/Carros Swift/Model/Utils/Prefs.swift
//
//  Prefs.swift
//  Carros Swift
//
//  Created by Antonio Lopes on 8/10/16.
//  Copyright © 2016 Antonio Lopes. All rights reserved.
//

import UIKit

class Prefs: NSObject {
    
    class func setObjectForKey(_ valor: NSObject, chave: String) {
        let prefs = UserDefaults.standard
        prefs.set(valor, forKey: chave)
        prefs.synchronize()
    }
    
    class func getObjectForKey(_ chave: String) -> AnyObject! {
        let prefs = UserDefaults.standard
        return prefs.object(forKey: chave) as AnyObject!
    }
}
